woocommerce/woocommerce
# Add modal editor for use in product editor

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Adds a modal editor that can be used in the product editor for descriptions and other fields that require separate editors.

This PR only addresses fitting the iframe editor into the editor and does not include the toolbar, sidebars, or other functionality which will be addressed in follow-ups.

<img width="1182" alt="Screen Shot 2023-04-21 at 1 29 02 PM" src="https://user-images.githubusercontent.com/10561050/233729977-082e7294-bed8-4d42-a4fb-08091d2fea69.png">


Closes #37898 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Navigate to Tools -> WCA Test Helper -> Features and enable the new product blocks editing experience
2. Navigate to Products -> Add new
3. Click on "Add description" under the description
4. Add some content via the blocks editor
5. Click the modal's close button ("x")
6. Make sure the description button now reads "Edit description"
7. Save the product
8. Refresh the page
9. Click "Edit description"
10. Make sure the description has persisted
